src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala

```diff
@@ -96,20 +96,25 @@ trait WorkflowSubmission extends FutureSupport with LazyLogging with MethodWiths
 
   //Get a blob of unlaunched workflows, flip their status, and queue them for submission.
   def getUnlaunchedWorkflowBatch()(implicit executionContext: ExecutionContext): Future[WorkflowSubmissionMessage] = {
-    val unlaunchedWfOptF = dataSource.inTransaction { dataAccess =>
+    val workflowRecsToLaunch = dataSource.inTransaction { dataAccess =>
       for {
         runningCount <- dataAccess.workflowQuery.countWorkflows(WorkflowStatuses.runningStatuses)
         reservedWorkflowRecs <- reserveWorkflowBatch(dataAccess, runningCount)
       } yield reservedWorkflowRecs
     }
 
-    unlaunchedWfOptF.map {
-      // submit the batch we found
-      case workflowRecs if workflowRecs.nonEmpty => SubmitWorkflowBatch(workflowRecs.map(_.id))
-
-      // if we didn't find any then look again in a little bit
-      case _ => ScheduleNextWorkflowQuery
+    //flip the workflows to Launching in a separate txn.
+    //if this optimistic-lock-exceptions with another txn, this one will barf and we'll reschedule when we pipe it back to ourselves
+    workflowRecsToLaunch map { wfRecs =>
+      dataSource.inTransaction { dataAccess =>
+        dataAccess.workflowQuery.batchUpdateStatus(wfRecs, WorkflowStatuses.Launching).map(_ => wfRecs)
```

Comment:
this has been moved up here from inside `reserveWorkflowBatch`, if you were looking for it